README.md

```diff
@@ -204,44 +204,64 @@ Prior to cloning the repository and attempting to build the project, ensure that
+#### Building the Python module using pip (this is what you probably want if you are not a developer)
+**NOTE**: this step is mandatory for setting up the Command Line Interface for Desbordante.
+
 Firstly, navigate to a desired directory.
-Then, clone the repository, cd into the project directory and launch the build script:
+Then, clone the repository, cd into the project directory and run the following commands:
 ```
-git clone https://github.com/Mstrutov/Desbordante/
-cd Desbordante
-./pull_datasets.sh
-./build.sh
+python3 -m venv venv
+source venv/bin/activate
+python3 -m pip install .
 ```
 
-### Building the project (second option: without tests)
+Now it is possible to `import desbordante` as a module from within the created virtual environment. The CLI for Desbordante can be run using the following commands:
+```sh
+pip install -r cli/requirements.txt
+python3 cli/cli.py --help
+```
 
-Firstly, navigate to a desired directory.
-Then, clone the repository, cd into the project directory and launch the build script:
+#### Building tests & the Python module manually
+In order to build tests, pull the test datasets using the following command:
+```sh
+./pull_datasets.sh
 ```
-git clone https://github.com/Mstrutov/Desbordante/
-cd Desbordante
-./build.sh --no-tests --no-unpack
+then build the tests themselves:
+```sh
+./build.sh -j$(nproc)
+```
+
+The Python module can be built by providing `--pybind` switch:
+```sh
+./build.sh --pybind -j$(nproc)
 ```
 
-### Launching the binaries
-The script generates the following file structure in `/path/to/Desbordante/build/target`:
+See `./build.sh --help` for more available options.
+
+The `./build.sh` script generates the following file structure in `/path/to/Desbordante/build/target`:
 ```bash
 ├───input_data
 │   └───some-sample-csv\'s.csv
 ├───Desbordante_test
-├───Desbordante_run
+├───desbordante.cpython-*.so
 ```
-The `input_data` directory contains several .csv files that may be used by `Desbordante_test`. Run `Desbordante_test` to perform unit testing:
+
+The `input_data` directory contains several .csv files that are used by `Desbordante_test`. Run `Desbordante_test` to perform unit testing:
 ```
 cd build/target
 ./Desbordante_test
 ```
-The tool itself may be run like the following:
+
+`desbordante.cpython-*.so` is a Python module, packaging Python bindings for the Desbordante core library. In order to use it, simply `import` it:
 ```
-./Desbordante_run --algo=tane --data=<path_to_dataset>
+cd build/target
+python3
+>>> import desbordante
 ```
 
+We use [easyloggingpp](https://github.com/abumq/easyloggingpp) in order to log (mostly debug) information in the core library. Python bindings search for a configuration file in the directory from which `desbordante` was imported, so to configure logging, create `logging.conf` in the directory desbordante will be imported from. This means that when running CLI `logging.conf` should be created in the directory from which `cli.py` is run.
```

Comment:
```suggestion
We use [easyloggingpp](https://github.com/abumq/easyloggingpp) in order to log (mostly debug) information in the core library. Python bindings search for a configuration file in the working directory, so to configure logging, create `logging.conf` in the directory from which desbordante will be imported. In particular, when running the CLI, `logging.conf` should be located in the directory from which the command to start it is to be issued.
```

Reply:
> the command to start it is to be issued.

I think it's better to be specific here

Reply:
Oh, is it expected to install `cli.py` to some directory in `$PATH`?

Reply:
Keeping the name `cli.py` is a pretty bad idea then.

Reply:
> > the command to start it is to be issued.
> 
> I think it's better to be specific here

The specific command is going to look like `/path/to/Desbordante/cli/cli.py` or `python3 /path/to/Desbordante/cli/cli.py` if it's not in `$PATH`

Reply:
> > Keeping the name cli.py is a pretty bad idea then.
> 
> why?

I added it a few seconds after. If it's in `$PATH`, it should have a distinct name. Any number of programs could be called `cli.py`, and there should be no ambiguity here.

Reply:
```suggestion
We use [easyloggingpp](https://github.com/abumq/easyloggingpp) in order to log (mostly debug) information in the core library. Python bindings search for a configuration file in the working directory, so to configure logging, create `logging.conf` in the directory from which desbordante will be imported. In particular, when running `python3 ./relative/path/to/cli.py`, `logging.conf` should be located in `.`.
```

Reply:
Oh, "`cli.py`" isn't the exact command, it's the file, right? Then the current version is fine, though some further emphasis would be nice ("is to be run." -> "is to be run, which is not necessarily the same directory it is located in.", if that's what is meant).